### DISCUSSION TITLE
Missing algorithm

### DISCUSSION THREAD

**Discussion by elifmeseci** [networkx/networkx#5909] [Algorithms]
There is no function related to Sorenson Index. How about adding Sorenson Index algorithm for link prediction?

**Comment by rossbar** [comment]
I'm no expert, but the impression that I got from the [wikipedia page](https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient) is that this seems more like a set operation than a network operation - AFAICT only nodes are considered in the calculation. If I've understood correctly, it should be straightforward to calculate the Sorensen index using set operations, e.g., assuming you have two graphs `G` and `H`:

```python
sorensen_ind = 2 * len( G.nodes | H.nodes ) / (len(G) + len(H))
```

If this is correct (please double-check!), then my personal preference would be that this calculation live in user code rather than as a function in NetworkX. In general I believe it would make user code more readable if this calcuation were explicit rather than "hiding behind" a named function `nx.sorensen_index`. That's just my opinion though, if there are other arguments for including a function please follow up here!
